code/mob/melee_attack_procs.dm

```diff
@@ -676,14 +676,16 @@
 			stam_power *= clamp(damage/pre_armor_damage, 1, 1/3)
 		stam_power *= max(0, (1-shield_amt["shield_strength"]))
 
-		//record the stamina damage to do
-		msgs.stamina_target -= max(stam_power, 0)
+		if (!target.hasStatus("weakened"))
+			//record the stamina damage to do
+			msgs.stamina_target -= max(stam_power, 0)
```

Comment:
?

Reply:
punches use this weird way to track their stamina damage.

Reply:
Yees, but why would being weakened make you immune to stamina damage? This isn't mentioned in the changelog.